pkg/storage/s3/s3.go

```diff
@@ -47,26 +57,82 @@ func NewDriver(c *imageregistryv1.ImageRegistryConfigStorageS3, kubeconfig *rest
 	}
 }
 
+// GetConfig reads configuration for the S3 cloud platform services.
+func GetConfig(kubeconfig *rest.Config, listers *regopclient.Listers) (*S3, error) {
+	cfg := &S3{}
+
+	infra, err := util.GetInfrastructure(listers)
+	if err != nil {
+		return nil, err
+	}
+
+	if infra.Status.PlatformStatus != nil && infra.Status.PlatformStatus.Type == configapiv1.AWSPlatformType {
+		cfg.Region = infra.Status.PlatformStatus.AWS.Region
+	}
+
+	client, err := coreset.NewForConfig(kubeconfig)
+	if err != nil {
+		return nil, err
+	}
+
+	// Look for a user defined secret to get the AWS credentials from first
+	sec, err := listers.Secrets.Get(imageregistryv1.ImageRegistryPrivateConfigurationUser)
+	if err != nil && errors.IsNotFound(err) {
+		// Fall back to those provided by the credential minter if nothing is provided by the user
+		sec, err = client.Secrets(imageregistryv1.ImageRegistryOperatorNamespace).Get(imageregistryv1.CloudCredentialsName, metav1.GetOptions{})
```

Comment:
As it's in the operator's namespace, you can use `listers.Secrets` here.

Reply:
If i remember correctly, we specifically did not use the listers cache here because we ran into an issue with it not being up-to-date maybe? let me see if i can hunt it down.

Reply:
@dmage Couldn't find it.  Updated to use the lister, if it passes tests it should be fine.